misc/windows-deploy/hostsetup.ps1

```diff
@@ -11,41 +11,119 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
+<#
+.NOTES
+This setup script is required in order to enable task IAM roles on windows instances.
+
+.NOTES
+169.254.170.2:51679 is the IP address used for task IAM roles.
+#>
+$CREDENTIAL_ADDRESS = "169.254.170.2"
+$CREDENTIAL_LISTEN_PORT = "80"
+$CREDENTIAL_CONNECT_PORT = "51679"
+
 $oldActionPref = $ErrorActionPreference
 $ErrorActionPreference = 'Continue'
 
-# This setup script is required in order to enable task IAM roles on windows instances.
-
-# 169.254.170.2:51679 is the IP address used for task IAM roles.
-$credentialAddress = "169.254.170.2"
-$credentialPort = "51679"
+$vmNetAdapter = Get-VMNetworkAdapter -SwitchName nat -ManagementOS -Name "*APIPA*" -ErrorAction:Ignore
 
-$adapter = (Get-NetAdapter -Name "*APIPA*")
-if(!($adapter)) {
-	# APIPA nat adapter controls IP range 169.254.x.x on windows.
-	Add-VMNetworkAdapter -VMNetworkAdapterName "APIPA" -SwitchName nat -ManagementOS
+if(!($vmNetAdapter)) {
+    # APIPA nat adapter controls IP range 169.254.x.x on windows.
+    Add-VMNetworkAdapter -VMNetworkAdapterName "APIPA" -SwitchName nat -ManagementOS
 }
+[int]$delay = 1
+[int]$maxTries = 10
+[bool]$available = $false
+While (-not $available -and $maxTries -ge 0) {
+    try {
+        $vmNetAdapter = Get-VMNetworkAdapter -SwitchName nat -ManagementOS -Name "*APIPA*" -ErrorAction:Ignore
+        if ($vmNetAdapter.count -ge 1) {
```

Comment:
If `vmNetAdapter` can be more than 1, what will happen on line 51 when you call the `MacAddress` method?